modules/Sources/Features/NHSendFlow/NHSendFlowStore.swift

```diff
@@ -0,0 +1,268 @@
+    @Dependency(\.derivationTool) var derivationTool
+    @Dependency(\.mainQueue) var mainQueue
+    @Dependency(\.mnemonic) var mnemonic
+    @Dependency(\.sdkSynchronizer) var sdkSynchronizer
+    @Dependency(\.walletStorage) var walletStorage
+    @Dependency(\.zcashSDKEnvironment) var zcashSDKEnvironment
+    
+    public var body: some ReducerProtocol<State, Action> {
+        BindingReducer()
+        
+        Reduce { state, action in
+            switch action {
+            case let .path(.element(id: id, action: .addMemo(.continueOrSkipTapped))):
+                guard case let .addMemo(addMemoState) = state.path[id: id]
+                else { return .none }
+                state.memo = addMemoState.memo
+                state.path.append(Path.State.recipient(.init()))
+                return .none
+            case .path(.element(id: _, action: .recipient(.continueTapped))):
+                guard derivationTool.isZcashAddress(state.recipient.data, networkType) else { return .none }
+                state.path.append(
+                    Path.State.review(
+                        .init(
+                            subtotal: state.amountToSend,
+                            memo: state.memo,
+                            recipient: state.recipient
+                        )
+                    )
+                )
+                return .none
+            case .path(.element(id: _, action: .recipient(.scanQRCodeTapped))):
+                state.path.append(Path.State.scan(.init()))
+                return .none
+            case let .path(.element(id: _, action: .scan(.found(code)))):
+                state.recipient = code
+                state.path = StackState(
+                    state.path.map { state in
+                        if case var .recipient(recipientState) = state {
+                            recipientState.recipient = code
+                            return Path.State.recipient(recipientState)
+                        }
+                    
+                        return state
+                    }
+                )
+                let _ = state.path.popLast()
+                return .none
+            case .path(.element(id: _, action: .review(.sendZcashTapped))):
+                guard !state.isSendingTransaction else { return .none }
+                
+                do {
+                    let storedWallet = try walletStorage.exportWallet()
+                    let seedBytes = try mnemonic.toSeed(storedWallet.seedPhrase.value())
+                    let spendingKey = try derivationTool.deriveSpendingKey(seedBytes, 0, networkType)
+                    
+                    state.isSendingTransaction = true
+                    
+                    let memo: Memo? = state.memo.data.isEmpty ? nil : try Memo(string: state.memo.data)
+                    
+                    let recipient = try Recipient(state.recipient.data, network: networkType)
+                    return .run { [state] send in
```

Comment:
[blocking] 

This code might be attempting to send a transaction to a transparent address with a memo which would fail.